wazo_dird/plugins/phonebook_backend/plugin.py

```diff
@@ -138,16 +136,14 @@ def _get_phonebook_key(
     ) -> database.PhonebookKey:
         if 'phonebook_uuid' in config:
             return database.PhonebookKey(uuid=config['phonebook_uuid'])
-        elif 'phonebook_id' in config:
-            return database.PhonebookKey(id=config['phonebook_id'])
-
-        phonebook_name = config['name']
-        phonebooks = self._crud.list([tenant_uuid], search=phonebook_name)
-        for phonebook in phonebooks:
-            if phonebook['name'] == phonebook_name:
-                return database.PhonebookKey(uuid=phonebook['uuid'])
-
-        raise InvalidConfigError(
-            f'sources/{self._source_name}/phonebook_name',
-            f'unknown phonebook {phonebook_name}',
-        )
+        else:
+            phonebook_name = config['name']
+            phonebooks = self._crud.list([tenant_uuid], search=phonebook_name)
+            for phonebook in phonebooks:
+                if phonebook['name'] == phonebook_name:
+                    return database.PhonebookKey(uuid=phonebook['uuid'])
```

Comment:
Did the source name allow matching with the phonebook name? If so, can we make a migration script to migrate sources with `name` but without `phonebook_uuid` to link them the correct `phonebook_uuid`, instead of simply deleting them?

Reply:
Yeah, this could be done. There's no guarantee that the source `name` matches the `phonebook.name`, but it seems to be a supported convention.